modules/qna/src/views/full/FormModal.jsx

```diff
@@ -39,20 +39,23 @@ export default class FormModal extends Component {
 
   state = this.defaultState
 
-  componentDidUpdate(prevProps) {
+  async componentDidUpdate(prevProps) {
     const { id } = this.props
     if (prevProps.id === id) {
       return
     }
     if (!id) {
       return this.setState(this.defaultState)
     }
-    this.props.bp.axios.get(`/mod/qna/questions/${id}`).then(({ data: { data: item } }) => {
-      this.setState({
-        item,
-        isRedirect: [ACTIONS.REDIRECT, ACTIONS.TEXT_REDIRECT].includes(item.action),
-        isText: [ACTIONS.TEXT, ACTIONS.TEXT_REDIRECT].includes(item.action)
-      })
+
+    const {
+      data: { data: item }
+    } = await this.props.bp.axios.get(`/mod/qna/questions/${id}`)
+
+    this.setState({
+      item,
+      isRedirect: [ACTIONS.REDIRECT, ACTIONS.TEXT_REDIRECT].includes(item.action),
+      isText: [ACTIONS.TEXT, ACTIONS.TEXT_REDIRECT].includes(item.action)
```

Comment:
instead of making componentDidUpdate async, I'd move the logic to its own method., like loadQuestion().

Reply:
Didn't write this part simply copy / paste it